### PR TITLE
Apply `opensafely-core/repo-template`

### PR DESCRIPTION
**Pull request by iaindillingham**
f81ab3b is the main commit in this pull request:

```
We apply the configuration found in the opensafely-core/repo-template
repo to this repo and replace the shell scripts in the bin directory
with a justfile. We make a small number of changes to the configuration,
most importantly setting the Python version to 3.8, which is that used
by opensafely-core/python-docker.

We upgrade dev and prod dependencies to ensure that everything looks
right (e.g. requirements.dev.txt and requirements.prod.txt contain
hashes). However, we expect to downgrade the prod dependencies, to
reflect opensafely-core/python-docker.

We remove some unnecessary files, such as .python-version and setup.cfg,
that we thought were necessary before we designed and implemented
reusable actions.
```

If you check out this branch and run `just test`, then you'll notice that the tests fail. I'm going to defer fixing them until @ccunningham101, @milanwiedemann, and I have talked about our plans for deciles-chart-action. (Besides, they fail even harder after we downgrade Pandas to the version used by [opensafely-core/python-docker](https://github.com/opensafely-core/python-docker) 😬.)

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+## System requirements
+
+### `just`
+
+```sh
+# macOS
+brew install just
+
+# Linux
+# Install from https://github.com/casey/just/releases
```
Comment (ccunningham101) [inline DEVELOPERS.md:12]: Manual download because the most recent version in pip is 0.8.138?

**Review by ccunningham101 — APPROVED** [review]
All the tooling changes make sense!

File: DEVELOPERS.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Notes for developers
+
+## System requirements
+
+### `just`
+
+```sh
+# macOS
+brew install just
```
Comment (milanwiedemann) [inline DEVELOPERS.md:9]: yay this worked for me

File: DEVELOPERS.md (hunk 1 above)
Comment (iaindillingham) [reply DEVELOPERS.md:12]: I didn't know `just` was on PyPI. I think it's probably a different command. Did you `pip install` [this one](https://pypi.org/project/just/), @ccunningham101? If so, then I'll need to make the documentation clearer. There are too many `just`s in the world.

File: DEVELOPERS.md (hunk 2 above)
Comment (iaindillingham) [reply DEVELOPERS.md:9]: This is the correct `just`. More here: https://github.com/casey/just.

File: DEVELOPERS.md (hunk 1 above)
Comment (ccunningham101) [reply DEVELOPERS.md:12]: Ah yes never mind, that is a different just!

**Comment by iaindillingham** [comment]
I've added a note about how the `just` we use isn't the `just` on PyPI. I've force-pushed to this branch, so if you checked it out, then it's probably best to check out `main` and then delete this branch:

```sh
git checkout main
git branch -D iaindillingham/apply-repo-template
```

**Pull request merged**